query/query.go

```diff
@@ -1654,13 +1664,13 @@ func (sg *SubGraph) recursiveFillVars(doneVars map[string]varValue) error {
 		return err
 	}
 	for _, child := range sg.Children {
-		err = child.recursiveFillVars(doneVars)
+		err := child.recursiveFillVars(doneVars)
```

Comment:
nit: if you are doing this, you can also move the statement inside the if. Same for some of the statements below.